DataAccessor/DataAccessor.cpp

```diff
@@ -0,0 +1,182 @@
+//
+// Created by Laura Kucharska on 28/11/2022.
+//
+// Implementation of DataAccessor class.
+// 
+//
+
+#include "DataAccessor.h"
+#include "wfdb.h"
+
+#include <filesystem>
+#include <iterator>
+#include <iostream>
+#include <algorithm>
+#include <fstream>
+#include <regex>
+
+inline std::string trim(std::string &str) {
+    return str.erase(str.find_last_not_of("\t\n\v\f\r ") + 1).erase(0, str.find_first_not_of("\t\n\v\f\r "));
+}
+
+void DataAccessor::parseComment() {
+    for (auto &line: comment) {
+        std::smatch match;
+        line = trim(line);
+        if (std::regex_search(line, match, std::regex(R"(^\d+\s[FM]\s)"))) {
+            std::string mathString = match.str(0);
+            auto spaceIndex = mathString.find_first_of(' ');
+            age = std::stoi(mathString.substr(0, spaceIndex));
+            sex = mathString.at(spaceIndex + 1);
+        } else if (std::regex_search(line, match, std::regex(R"(^Age:\s+\d+\s+Sex:\s+[FM])"))) {
+            std::string mathString = match.str(0);
+            std::regex_search(line, match, std::regex(R"(\d+)"));
+            age = std::stoi(match.str(0));
+            std::regex_search(line, match, std::regex(R"([FM])"));
+            sex = match.str(0).at(0);
+        }
+    }
+}
+
+bool DataAccessor::load(std::string filepath) {
+    m_loaded = false;
+    m_signalCount = 0;
+    namespace fs = std::filesystem;
+
+    // Windows can deal with '/' as a directory separator.
+    std::replace(filepath.begin(), filepath.end(), '\\', '/');
+
+    int splitPoint = filepath.find_last_of(m_separator);
+    std::string directoryPath = filepath.substr(0, splitPoint);
+    std::string fileNameWithExtension = filepath.substr(splitPoint + 1);
+    m_fileNameWithoutExtension = fileNameWithExtension.substr(0, fileNameWithExtension.find_first_of('.'));
+
+    // Checking .dat file
+    std::string datFilePath = directoryPath + m_separator + m_fileNameWithoutExtension + ".dat";
+    if (!fs::exists(datFilePath)) {
+        return false;
+    }
+
+    // Checking .hea file
+    m_heaFilePath = directoryPath + m_separator + m_fileNameWithoutExtension + ".hea";
```

Comment:
To zachowanie jest niebezpieczne. Nie możemy oczekiwać, że zawsze te pliki będą znajdowały się w jednym katalogu. Ustaliliśmy, że na wejście `DataAcessor` przyjdą dwa `std::string`: jeden do pliku `.dat` drugi do pliku `.hea`. Zmień proszę implementację tak aby uwzględnić nowy argument.

Reply:
Niestety biblioteka biblioteka wfdb (to co z niej skopiowałam) opiera się na tym, że plik `.hea` i `.dat` znajduje się w tym samym katalogu. Głównie dlatego, że w headerze jest informacja jak nazywa się plik z danymi (lub nazywaj się, bo może być ich więcej). Więc tak na prawdę to powinniśmy wskazywać tylko na header.